client/dashboard/data/domain-transfer.ts

```diff
@@ -0,0 +1,53 @@
+import wpcom from 'calypso/lib/wp';
+
+export type IpsTag = {
+	tag: string;
+	registrarName: string;
+	registrarUrl: string;
+};
+
+export async function updateDomainLock( domain: string, enabled: boolean ): Promise< void > {
+	return wpcom.req.post( {
+		path: `/domains/${ domain }/transfer`,
+		body: {
+			domainStatus: JSON.stringify( {
+				command: enabled ? 'lock' : 'unlock',
+			} ),
+		},
+	} );
+}
+
+export async function requestTransferCode( domain: string ): Promise< void > {
+	return wpcom.req.post( {
+		path: `/domains/${ domain }/transfer`,
+		body: {
+			domainStatus: JSON.stringify( {
+				command: 'only-send-code',
+			} ),
+		},
+	} );
+}
+
+export async function saveIpsTag( domain: string, ipsTag: string ): Promise< void > {
+	return wpcom.req.post( {
+		path: `/domains/${ domain }/transfer`,
+		body: {
+			domainStatus: JSON.stringify( {
+				command: 'set-ips-tag',
+				payload: { ips_tag: ipsTag },
+			} ),
+		},
+	} );
+}
+
+export async function getIpsTagList(): Promise< IpsTag[] > {
+	try {
+		const response = await fetch( 'https://widgets.wp.com/domains/ips-tag-list.min.json' );
+		if ( ! response.ok ) {
+			throw new Error( `HTTP error! status: ${ response.status }` );
```

Comment:
Basically all the exceptions thrown by our data functions are `WPError`, since it is what the `wpcom` library throws. An advantage of having this function throw a `WPError` too is that calls will know how to extract the status code.

The code `wpcom` uses for creating errors from `! ok` responses here:
https://github.com/Automattic/wp-calypso/blob/a2e34f0bfb73603e65a84076c8c2ae379e6b8d29/packages/wpcom-xhr-request/src/index.js#L90